.gitignore

```diff
@@ -111,3 +111,5 @@ install_manifest.txt
 # vim 
 *.swp
 *.swo
+
+docs
```

Comment:
We don't plan to use doxygen for source code documentation. @blair1618 had mentioned his inclination towards http://jessevdk.github.io/cldoc/.